File: src/my_project_typescript/easy_problems_typescript/from1to50/minimumAbsoluteDifference.ts
function minimumAbsDifference(arr: number[]): number[][] {
    let answer: any  = []

    arr.sort((a,b) => a <= b?-1:1)

    let minDistance: number = Infinity
    let temp: number = 0

    for (let i = 0; i < arr.length-1; i++){
        if (arr[i+1] - arr[i] <= minDistance){
            minDistance = arr[i+1] - arr[i]
            if (temp > minDistance){
                answer = []
            }
            temp = minDistance
            answer.push([arr[i], arr[i+1]])
        }
    }
    return answer
};

// function to be reverted 
function myFunction(myVariable:  number): number{
    return myVariable*7
}

console.log(minimumAbsDifference([4,2,1,3]))

console.log(minimumAbsDifference([1,3,6,10,15]))


console.log(minimumAbsDifference([3,8,-10,23,19,-4,-14,27]))

File: src/my_project/easy_problems/from101to150/minimum_absolute_difference.py
from typing import List 
from math import inf 

class SecondSolution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:

        arr.sort()

        list_pairs = list()
        len_arr = len(arr)
        for i in range(len_arr):
            for j in range(i+1, len_arr):
                list_pairs.append([arr[i], arr[j]])

        list_pairs.sort(key= lambda x: abs(x[0]-x[1]))

        list_pairs = [x for x in list_pairs if abs(x[0] - x[1]) == abs(list_pairs[0][0] - list_pairs[0][1])]

        return list_pairs


class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:

        answer = list()
        arr.sort()
        min_distance = inf 
        temp = 0

        for i in range(len(arr)-1):
            if arr[i+1] - arr[i] <= min_distance:
                min_distance = arr[i+1] - arr[i]
                if temp > min_distance :
                    answer.clear()
                temp = min_distance
                answer.append([arr[i], arr[i+1]])
        


        return answer 


solution = Solution()

print(solution.minimumAbsDifference(arr = [4,2,1,3]))

print(solution.minimumAbsDifference(arr = [1,3,6,10,15]))


print(solution.minimumAbsDifference(arr = [3,8,-10,23,19,-4,-14,27]))



